### PR TITLE
Abort PKC config without Docker Engine

### LINKED ISSUE #59: Check if docker is running before running the pkc setup script

### PR DESCRIPTION
**Pull request by 0xc0000007b**
closes #59

### CODE REVIEW

File: pkc [hunk 1]
```diff
@@ -54,26 +54,26 @@ check_wallet_address() {
 
 check_dependencies() {
     # Check Docker installation
-    if [[ -x "$(command -v docker)" ]]; then
-        true
-    else
-        echo "Docker is not yet installed. Aborting PKC installation."
+    if [[ ! -x "$(command -v docker)" ]]; then
```
Comment (rht) [inline pkc:57]: Lol thanks for cleaning the mess!

File: pkc [hunk 2]
```diff
@@ -54,26 +54,26 @@ check_wallet_address() {
...
-        true
-    else
-        echo "Git is not yet installed. Aborting PKC installation."
+    if [[ ! -x "$(command -v git)" ]]; then
+        echo "Error: Git is not yet installed. Aborting PKC installation." >&2
+        exit 1
+    fi
+
+    # Check Docker Engine running
+    if [[ $(docker version) != *"Engine"* ]]; then
```
Comment (rht) [inline pkc:75]: I tested this by doing:
- `sudo systemctl stop docker` also `sudo systemctl disable docker`
- `docker version`
It outputs the text `Engine` in both the `stop` and `disable` cases.
It seems this check is macOS-specific? Also, shouldn't the goal be detecting whether the Docker daemon is both enabled and running?

File: pkc [hunk 3]
```diff
@@ -54,26 +54,26 @@ check_wallet_address() {
 
 check_dependencies() {
     # Check Docker installation
-    if [[ -x "$(command -v docker)" ]]; then
-        true
-    else
-        echo "Docker is not yet installed. Aborting PKC installation."
+    if [[ ! -x "$(command -v docker)" ]]; then
+        echo "Error: Docker is not yet installed. Aborting PKC installation." >&2
```
Comment (rht) [inline pkc:58]: :100:

File: pkc (hunk 2 above)
Comment (0xc0000007b) [reply pkc:75]: thanks for testing on linux. sounds like it is indeed environment specific. will look for an alternative so we don't have to fork code based on platform

**Comment by rht** [comment]
Merged :tada:

File: pkc [hunk 4]
```diff
@@ -53,29 +53,47 @@ check_wallet_address() {
...
+        local i=1
+        local sp="/-\|"
+        echo -n ' '
+        until [[ "$(docker version 2>/dev/null)" == *"Engine"* ]]; do
+            # shellcheck disable=SC2059
+            printf "\b${sp:i++%${#sp}:1}"
+            sleep 1
+        done
+        echo ""
+    fi
```
Comment (rht) [inline pkc:96]: Ah, forgot to echo messages for the case where neither systemd nor launchd exists in the system (e.g. Gentoo / BSDs); @0xc0000007b

**Pull request merged**